CMake/ctkMacroBuildQtPlugin.cmake

```diff
@@ -181,12 +181,23 @@ macro(ctkMacroBuildQtPlugin)
 endmacro()
 
 macro(ctkMacroBuildQtDesignerPlugin)
+  if(CTK_QT_VERSION VERSION_GREATER "4")
+    find_package(Qt5Designer REQUIRED)
+    add_definitions(${Qt5Designer_DEFINITIONS})
+    include_directories(${Qt5Designer_INCLUDE_DIRS})
+  endif()
   ctkMacroBuildQtPlugin(
     PLUGIN_DIR designer
     ${ARGN})
+  cmake_parse_arguments(MY
+    "" # no options
+    "NAME;EXPORT_DIRECTIVE;FOLDER;PLUGIN_DIR" # one value args
+    "SRCS;MOC_SRCS;UI_FORMS;INCLUDE_DIRECTORIES;TARGET_LIBRARIES;RESOURCES" # multi value args
+    ${ARGN}
+    )
   if(CTK_QT_VERSION VERSION_GREATER "4")
     find_package(Qt5Designer REQUIRED)
-    target_link_libraries(${lib_name} Qt5::Designer)
+    qt5_use_modules(${MY_NAME} Designer)
```

Comment:
The target_link_libraries was not the issue. However, the moc-ing of the ctkWidgetsPlugin.cpp was failing because some Qt Designer headers where missing.
qt5_use_modules was adding those headers. 
But I can do instead 
```
add_definitions(${Qt5Designer_DEFINITIONS})
include_directories(${Qt5Designer_INCLUDE_DIRS})
```
and leave the ```target_link_libraries```